wombat-admin/src/app/page.tsx

```diff
@@ -34,51 +42,29 @@ export default function Home() {
     }
   };
 
-  const [activeTab, setActiveTab] = useState<string>('Daily Visits');
 
+  // This function allows to navigate between Daily Visits and User Audit Trail tabs
   const handleTabClick = (tabName: string) => {
     setActiveTab(tabName);
   };
 
+  // Custom hook for fetching requests data
+  const requestsData = useDataFetcher('http://localhost:4040/access_request');
 
-  const [requests, setRequests] = useState([]);
+  // Custom hook for fetching users data
+  const usersData = useDataFetcher('http://localhost:4040/users');
 
+  // Update state based on fetched data
   useEffect(() => {
-    const fetchData = async () => {
-      try {
-        const response = await fetch(`http://localhost:4040/access_request`);
-        if (!response.ok) {
-          throw new Error('Failed to fetch data');
-        }
-        const data = await response.json();
-        setRequests(data); // Assuming API returns an array of requests
-      } catch (error) {
-        console.error('Error fetching data:', error);
-      }
-    };
-    fetchData();
-  }, []);
-
-
-  const [users, setUsers] = useState([]);
+    setRequests(requestsData);
+  }, [requestsData]);
 
   useEffect(() => {
```

Comment:
You should not need useEffect here anymore I don't think. Its been moved into the hook.